flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterClient.java

```diff
@@ -211,18 +185,45 @@ public void run() {
 		Runtime.getRuntime().addShutdownHook(clientShutdownHook);
 
 		isConnected = true;
+
+		logAndSysout("Waiting until all TaskManagers have connected");
```

Comment:
Thank you. I fixed that by using a non-static variable for the logger and dynamically retrieving the class name, i.e.
```java
	private final Logger LOG = LoggerFactory.getLogger(getClass());
```